server/src/main/java/org/apache/druid/server/QueryResource.java

```diff
@@ -230,8 +230,7 @@ public void write(OutputStream outputStream) throws WebApplicationException
                       // json serializer will always close the yielder
                       jsonWriter.writeValue(os, yielder);
 
-                      os.flush(); // Some types of OutputStream suppress flush errors in the .close() method.
-                      os.close();
+                      os.flush();
```

Comment:
maybe we should also do a flush in both QueryResource and SqlResource in the `finally` block. That way, the last line is going to be a complete one.